program-test/tests/warp.rs

```diff
@@ -209,11 +134,30 @@ async fn clock_sysvar_updated_from_warp() {
     );
 }
 
+async fn check_credits_observed(
+    banks_client: &mut BanksClient,
+    stake_address: Pubkey,
+    expected_credits: u64,
+) {
+    let stake_account = banks_client
+        .get_account(stake_address)
+        .await
+        .unwrap()
+        .unwrap();
+    let stake_state: StakeState = deserialize(&stake_account.data).unwrap();
+    assert_eq!(
+        stake_state.stake().unwrap().credits_observed,
+        expected_credits
+    );
+}
+
 #[tokio::test]
 async fn stake_rewards_from_warp() {
     // Initialize and start the test network
     let program_test = ProgramTest::default();
     let mut context = program_test.start_with_context().await;
+
+    context.warp_to_slot(100).unwrap();
```

Comment:
should _really_ avoid mixing functional and cosmetic changes like this. it makes review very error prone. i spent 10min trying to figure out what this addition was for before noticing that it was pulled out of `setup_vote()`

Reply:
yeah. 
Need to move wrap out so that we can call setup mutiple vote accounts. To keep the original caller to have the same behavior, those "warp" are inserted.

Reply:
To be clear, Trent isn't objecting to the change, just requesting that the move and the edit be noted in separate commits. So the commits in this PR might be something like:
1. Add `deactivate_stake()` helper
2. Return error when feature activated and StakeFlag set
3. Add stake_flag unit test
4. Move `setup_stake` to its own mod
5. Only warp in warp.rs tests
6. Add stake.rs tests

I don't want to bog this down in a reorg, so just a consideration for next time.